### PR TITLE
Migrate enzyme Button tests to React Testing Library

### PR DESCRIPTION
**Pull request by b3h3m0th**
#### What's in this PR?

Button test migrations from enzyme to RTL

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Button/tests/__snapshots__/Button.test.js.snap [hunk 1]
```diff
@@ -1,129 +1,222 @@
...
+        type="button"
+      >
+        <span
+          aria-label="su-angle-down"
+          class="su-angle-down dropdownIcon"
+        />
+      </button>
+    </div>
+  </body>,
+  "container": <div>
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Button/tests/__snapshots__/Button.test.js.snap]: This looks strange look like we are creating here a false snapshot. I think the snapshot is done on the false element.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Button/tests/Button.test.js [hunk 2]
```diff
@@ -1,72 +1,90 @@
...
 });
 
 test('Should render with skin secondary and dropdown icon', () => {
-    expect(render(<Button showDropdownIcon={true} skin="secondary" />)).toMatchSnapshot();
+    const {container} = render(<Button showDropdownIcon={true} skin="secondary" />);
+    expect(container).toMatchSnapshot();
 });
 
 test('Should render with skin link and dropdown icon', () => {
     expect(render(<Button showDropdownIcon={true} skin="link" />)).toMatchSnapshot();
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Button/tests/Button.test.js]: the snapshot of this tests looks false. I think in react testing library we should not do this on test().toSnapshot instead on render()

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
@b3h3m0th  One small test need to be adopted. Else good work 👍

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Button/tests/__snapshots__/Button.test.js.snap (hunk 1 above)
Comment (b3h3m0th) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Button/tests/__snapshots__/Button.test.js.snap]: You're right. I forgot to migrate the [test you mentioned here](https://github.com/sulu/sulu/pull/6721#discussion_r931481107) which results in printing the whole `render()` return value to the snapshot file.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Button/tests/Button.test.js (hunk 2 above)
Comment (b3h3m0th) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Button/tests/Button.test.js]: See [this previous comment](https://github.com/sulu/sulu/pull/6721#discussion_r931492355).

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Button/tests/__snapshots__/Button.test.js.snap (hunk 1 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Button/tests/__snapshots__/Button.test.js.snap]: Yeah think moving the snapshot inside the `test` like its done in the other tests should fix it :)

**Review by alexander-schranz — APPROVED** [review]

**Comment by alexander-schranz** [comment]
@b3h3m0th Thx good work 👍

**Pull request merged**